warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java

```diff
@@ -0,0 +1,431 @@
+        
+        if (isValid && isParen(node) && isParen(grandChild(node))) {
+            // found a double paren
+            invalidate(node, "double paren");
+            return data;
+        } else if (!isValid) {
+            return data;
+        }
+        
+        node.childrenAccept(this, data);
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTOrNode node, Object data) {
+        if (isValid && isAnyChildAWrappedSingleTerm(node)) {
+            // found a single term child that is wrapped
+            invalidate(node, "wrapped single child");
+            return data;
+        }
+        node.childrenAccept(this, data);
+        return data;
+    }
+    
+    @Override
+    public Object visit(ASTAndNode node, Object data) {
+        if (!isValid)
+            return data;
+        
+        QueryPropertyMarker.Instance instance = QueryPropertyMarker.findInstance(node);
+        if (instance.isAnyType()) {
+            // descend into the marker's source node
+            JexlNode source = instance.getSource();
+            source.childrenAccept(this, data);
+            return data;
+        } else if (isAnyChildAWrappedSingleTerm(node)) {
+            // invalidate
+            invalidate(node, "wrapped single child");
+            return data;
+        } else {
+            // otherwise continue descent
+            node.childrenAccept(this, data);
+            return data;
+        }
+    }
+    
+    /*
+     * Helper methods
+     */
+    
+    // is this node a parentheses
+    private boolean isParen(JexlNode node) {
+        return node instanceof ASTReference && node.jjtGetNumChildren() == 1 && node.jjtGetChild(0) instanceof ASTReferenceExpression;
+    }
+    
+    // fetch the grandchild
+    private JexlNode grandChild(JexlNode node) {
+        return node.jjtGetChild(0).jjtGetChild(0);
+    }
+    
+    private boolean isAnyChildAWrappedSingleTerm(JexlNode node) {
+        for (int i = 0; i < node.jjtGetNumChildren(); i++) {
+            JexlNode child = node.jjtGetChild(i);
+            if (isWrappedSingleTerm(child)) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    private boolean isWrappedSingleTerm(JexlNode node) {
+        if (isParen(node)) {
+            boolean isMarkerNode = QueryPropertyMarker.findInstance(node).isAnyType();
+            if (!isMarkerNode) {
+                JexlNode unwrapped = JexlASTHelper.dereference(node);
+                
+                return !(unwrapped instanceof ASTAndNode || unwrapped instanceof ASTOrNode);
+            }
+        }
+        return false;
+    }
+    
+    private void invalidate(JexlNode node, String cause) {
+        isValid = false;
+        log.info("Invalid node (" + cause + "): " + JexlStringBuildingVisitor.buildQueryWithoutParse(node));
+    }
+    
+    /*
+     * Pass through
+     */
+    
+    @Override
+    public Object visit(ASTJexlScript node, Object data) {
```

Comment:
These are the same implementation as the parent class. These can just be removed I think.

Reply:
Works for me. I'll leave the comment in for which nodes are explicitly pass-through

Reply:
Actually, I'll remove the visit for ASTJexlScript and modify the others to only continue recursing through the tree if it's still valid. Opportunity to short circuit on reference expressions, simple nodes, and not nodes

Reply:
Cool. Sounds good.